catroid/src/androidTest/java/org/catrobat/catroid/test/content/actions/DeleteLookActionFileCleanupTest.kt

```diff
@@ -0,0 +1,145 @@
+    @Before
+    fun setUp() {
+        this.createTestProject()
+    }
+
+    @After
+    @Throws(Exception::class)
+    fun tearDown() {
+        TestUtils.deleteProjects(projectName)
+    }
+
+    @Test
+    fun testDeleteLook() {
+        val lookFileNamesList = mutableListOf("LookFile1", "LookFile2", "LookFile3")
+        addLookDataFromListToSprite(lookFileNamesList)
+
+        deleteNumberOfLooksAndAssert(1, lookFileNamesList)
+
+        ProjectLoadTask.task(project.directory, ApplicationProvider.getApplicationContext())
+        assertImageDirectoryOnlyContainsUsedFiles(lookFileNamesList)
+    }
+
+    @Test
+    fun testDeleteMultipleLooks() {
+        val lookFileNamesList = mutableListOf("LookFile1", "LookFile2", "LookFile3")
+        addLookDataFromListToSprite(lookFileNamesList)
+
+        deleteNumberOfLooksAndAssert(3, lookFileNamesList)
+        ProjectLoadTask.task(project.directory, ApplicationProvider.getApplicationContext())
+        assertImageDirectoryOnlyContainsUsedFiles(lookFileNamesList)
+    }
+
+    @Test
+    fun testNoFilesDeletedAfterLoadWithoutDelete() {
+        val lookFileNamesList = mutableListOf("LookFile1", "LookFile2", "LookFile3")
+        addLookDataFromListToSprite(lookFileNamesList)
+
+        deleteNumberOfLooksAndAssert(0, lookFileNamesList)
+
+        ProjectLoadTask.task(project.directory, ApplicationProvider.getApplicationContext())
+        ProjectLoadTask.task(project.directory, ApplicationProvider.getApplicationContext())
+        assertImageDirectoryOnlyContainsUsedFiles(lookFileNamesList)
+    }
+
+    private fun deleteNumberOfLooksAndAssert(
+        numberOfLooksToDelete: Int,
+        lookNameList: MutableList<String>
+    ) {
+        assertEquals(lookNameList.size, sprite.lookList.size)
+        assertImageDirectoryOnlyContainsUsedFiles(lookNameList)
+        for (numberOfDeletedLooks in 1..numberOfLooksToDelete) {
+            sprite.actionFactory.createDeleteLookAction(sprite).act(1f)
+            assertEquals(lookNameList.size - numberOfDeletedLooks, sprite.lookList.size)
+            assertImageDirectoryOnlyContainsUsedFiles(lookNameList)
+        }
+        repeat(numberOfLooksToDelete) { lookNameList.remove(lookNameList.first()) }
+    }
+
+    private fun assertImageDirectoryOnlyContainsUsedFiles(lookFileNamesList: List<String>) {
+        val imageDirectory = File(scene.directory, Constants.IMAGE_DIRECTORY_NAME)
+        val nomediaOffset = 1
+        assertEquals(lookFileNamesList.size + nomediaOffset, imageDirectory.listFiles().size)
+        lookFileNamesList.forEach { name ->
+            val file = File(imageDirectory, name)
+            assertTrue(file.exists())
+        }
+    }
+
+    private fun createTestProject() {
+        project = Project(ApplicationProvider.getApplicationContext(), projectName)
+        scene = project.defaultScene
+        sprite = Sprite("testSprite")
+        scene.addSprite(sprite)
+
+        XstreamSerializer.getInstance().saveProject(project)
```

Comment:
Since we don't have to test if StorageOperations.deleteFile(file) really deletes a file,
consider Mockito to test your implementation :)